### PR TITLE
Controllers spec

### PR DESCRIPTION
**Pull request by JacintaNgwu**
- Add RSpec as a gem to the project.
- Create a Request spec file for all controllers.
- For each action I checedk:
- [ ] If response status was correct.
- [ ] If a correct template was rendered.
- [ ] If the response body includes correct placeholder text.

### CODE REVIEW

File: spec/requests/posts_spec.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    it 'Should returns http success status' do
+      expect(response).to have_http_status(:ok)
+    end
+
+    it 'Should response includes placeholder text' do
+      expect(response.body).to include('<h1>Here is a list of all the posts</h1>')
+    end
+  end
+end
```
Comment (BenMukebo) [inline spec/requests/posts_spec.rb]: - [x] Great job so far 👏🏻 However the Post `show` actions tests should include tests that ensure the correct view template is rendered. please add this ✍🏻 Example : `expect(response).to render_template(:show)`

File: spec/requests/users_spec.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+    it 'returns a success response' do
+      get user_path(id: user.id)
+      expect(response).to be_successful
+    end
+
+    it 'renders the show template' do
+      get user_path(id: user.id)
+      expect(response).to render_template(:show)
+    end
```
Comment (BenMukebo) [inline spec/requests/users_spec.rb]: - [x] The Spec file should include tests that ensure the response body includes the correct placeholder. As you did on the pots-spec file Kindly make sure the `index` and `show` actions for the `User` spec include tests that ensure the response body includes the correct placeholder text. Example : `expect(response.body).to include("Welcome to user index page")`

**Review by BenMukebo — CHANGES_REQUESTED** [review]
### Hi @JacintaNgwu,

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

### To highlight 💪🏻
- Gitflow was correctly used. ✔️
- Professional Pull Request. ✔️
 👍  Good job!


## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by tufoinnkuo10 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @JacintaNgwu  

Your project is complete! Great job  🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Good Points 👍
- Descriptive Pull Request
- correct git flow
- No linter errors

<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**